pkg/trace/api/api.go

```diff
@@ -762,3 +763,34 @@ func getMediaType(req *http.Request) string {
 	}
 	return mt
 }
+
+// getDirector returns a director for a httputil.ReverseProxy. It enriches requests with headers setting custom
+// container and additional tags and overwrites the user agent header.
+func getHeaderDecoratorDirector(tags string, metricSuffix string) func(req *http.Request) {
+	return func(req *http.Request) {
+		req.Header.Set("Via", fmt.Sprintf("trace-agent %s", info.Version))
+		if _, ok := req.Header["User-Agent"]; !ok {
+			// explicitly disable User-Agent so it's not set to the default value
+			// that net/http gives it: Go-http-client/1.1
+			// See https://codereview.appspot.com/7532043
+			req.Header.Set("User-Agent", "")
+		}
+		containerID := req.Header.Get(headerContainerID)
+		if ctags := getContainerTags(containerID); ctags != "" {
+			req.Header.Set("X-Datadog-Container-Tags", ctags)
+		}
+		req.Header.Set("X-Datadog-Additional-Tags", tags)
+		_ = metrics.Count("datadog.trace_agent."+metricSuffix, 1, nil, 1)
+	}
+}
+
+// getOrchestrator returns the fargate orchestrator. Should only be called if the agent is running in ECS or EKS.
+func getOrchestrator() string {
```

Comment:
Why not call it `fargateOrchestratorTag` (I do prefer names to be as short as possible but I think we need to be specific here) and return the value `"orhcestrator:"+strings.ToLower(orch)` since this is what happens in both places. That would then be even more de-duplication.